site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -112,4 +120,130 @@ public function showSummary() {
         $rows = $this->core->getQueries()->getGradeables($gradeable_id, $student_ids, $section_key);
         $this->core->getOutput()->renderOutput(array('grading', 'ElectronicGrader'), 'summaryPage', $gradeable, $rows, $graders);
     }
-}
\ No newline at end of file
+
+    public function submitGrade() {
+        if (!isset($_POST['csrf_token']) || $_POST['csrf_token'] != $this->core->getCsrfToken()) {
+            $_SESSION['messages']['error'][] = "Invalid CSRF Token";
+            $this->core->redirect($this->core->buildUrl(array()));
+        }
+
+        $gradeable_id = $_POST['g_id'];
+        $who_id = $_POST['u_id'];
+        $gradeable = $this->core->getQueries()->getGradeable($gradeable_id, $who_id);
+
+        $now = new \DateTime('now');
+        $homeworkDate = $gradeable->getGradeStartDate();
+        if ($now < $homeworkDate) {
+            $_SESSION['messages']['error'][] = "Grading is not open yet for {$gradeable->getName()}";
+            $this->core->redirect($this->core->buildUrl(array('component'=>'grading', 'page'=>'electronic', 'gradeable_id' => $gradeable_id)));
+        }
+        
+        $regrade = $gradeable->beenTAgraded();
+        
+        $submit_data = array();
```

Comment:
Can you make an issue in Github (use the label 'refactor') that says "Refactor ElectronicGraderController:: submitGrade to use Gradeable model" or something like that, and then put a TODO above that points to that issue with its title. This works for now, but I'd like to start better tracking the various refactor tasks we've got.